src/run_all.py

```diff
@@ -0,0 +1,164 @@
+import os
+import sys
+import logging
+import time
+import getopt
+from urllib.parse import urlparse
+
+from applytemplate import main
+from listhosts import RancherInstances
+from listcontainers import Discover, getKey
+from image_checker import ImageChecker
+
+log = logging.getLogger(__name__)
+
+def run_apply_template(page, image_checker, dry_run):
+    assert page, "Please provide a template"
+
+    config = dict(
+        factsheet_project=os.getenv("FACTSHEET_PROJECT", "infrastructure"),
+        template_project=os.getenv("TEMPLATE_PROJECT", "netpub"),
+        template_name=os.getenv("TEMPLATE_NAME", "IT_service_factsheet_template"),
+        todolist_name=os.getenv("TODOLIST_NAME", "IT_service_factsheet_ToDo_list"),
+        wiki_server=os.getenv("WIKI_SERVER", ""),
+        wiki_apikey=os.getenv("WIKI_APIKEY", ""),
+        stackwiki=os.getenv("WIKI_STACKS_PAGE", "Rancher_stacks"),
+        dry_run=dry_run,
+    )
+
+    assert config["wiki_server"], "Please set WIKI_SERVER env var"
+    assert config["wiki_apikey"], "Please set WIKI_APIKEY env var"
+
+    main(page, config, image_checker)
+
+
+def run_list_hosts(dry_run, environments):
+    obj = RancherInstances(environments)
+    if obj.has_changed():
+        if dry_run == True:
+            obj.write_stdout()
+        else:
+            obj.write_page()
+    else:
+        logging.info("Content is the same, not saving")
+    logging.info("Done list hosts")
+
+def run_list_containers(image_checker, dry_run):
```

Comment:
Please don't duplicate all this code. Rather, in `list_containers.py`, extract it as a `main` function, and call that from here.

```python
def main():
    pageTitle = os.getenv('WIKI_CONTAINERSPAGETITLE', 'Rancher Containers')
    # ...

if __name__ == '__main__':
    main()
```

Same for the other "main" chunks of code.